### PR TITLE
Make `crypto share` additive

### PR DESCRIPTION
**Pull request by michaeljguarino**
## Summary

We'd forcibly overwrite all identities when sharing previously, which can cause weird issues.  This should ensure that we only add to the share list.

## Test Plan
tested on the plural-az-installations repo

## Checklist

- [x] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [x] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: pkg/crypto/age.go [hunk 1]
```diff
@@ -117,9 +117,20 @@ func SetupAge(client api.Client, emails []string) error {
 			return err
 		}
 
+		dedupeKey := func(id *AgeIdentity) string { return fmt.Sprintf("%s::%s", id.Email, id.Key) }
+
 		idents := ageConfig.Identities
+		present := map[string]bool{}
```
Comment (zreigz) [inline pkg/crypto/age.go:123]: "k8s.io/apimachinery/pkg/util/sets"

present := sets.sets.NewString

Comment (zreigz) [reply pkg/crypto/age.go:123]:
```
		present := sets.NewString()
		for _, id := range idents {
			present.Insert(dedupeKey(id))
		}

		for _, key := range keys {
			id := &AgeIdentity{Key: key.Content, Email: key.User.Email}
			if !present.Has(dedupeKey(id)) {
				idents = append(idents, id)
			}
			
		}

```

Comment (michaeljguarino) [reply pkg/crypto/age.go:123]: oh cool, i was hoping someone would build this

**Comment by michaeljguarino** [comment]
@zreigz used the sets package

**Pull request merged**